### DISCUSSION TITLE
请问如何使用主线uboot刷入固件？

### DISCUSSION THREAD

**Discussion by mouyase** [immortalwrt/immortalwrt#1308] [Q&A] [answered]
我手上的机器是Q30 Pro，想刷主线固件，结果发现似乎需要用主线uboot，而主线uboot又没有webui，所以不知道如何正确刷入固件，希望有大大可以贴出一个步骤或者教程。

**Comment by mouyase** [comment, marked as answer]
一、路由器开启SSH
不同路由器有不同的方法，需要自行寻找方案
二、刷入UBOOT
电脑连到路由器LAN口，用SSH连接路由器，上传`uboot.fip`和`preloader.bin`到tmp目录下
运行命令
```
mtd write /tmp/preloader.bin BL2
mtd write /tmp/uboot.fip FIP
```
等待1分钟后关机
三、安装基础Linux内核
安装Tftpd64，将`immortalwrt-xxxx-initramfs-recovery.itb`复制到Tftpd64根目录
电脑设置IP：192.168.1.254
路由器按住重置键不放，上电开机，直到看到Tftpd64有数据被读取
四、升级ImmortalWrt固件
SSH连接192.168.1.1
root/密码空
上传sysupgrade.itb到tmp目录下
运行命令 sysupgrade -v /tmp/sysupgrade.itb

祝您成功！

**Comment by liyiranlab** [comment]
https://github.com/immortalwrt/immortalwrt/discussions/1257
